test/public/lhcPeriods/overview.test.js

```diff
@@ -75,32 +75,29 @@ module.exports = () => {
 
     it('Should display the correct items counter at the bottom of the page', async () => {
         await goToPage(page, 'lhc-period-overview');
-        await waitForTimeout(100);
 
-        expect(await page.$eval('#firstRowIndex', (element) => parseInt(element.innerText, 10))).to.equal(1);
-        expect(await page.$eval('#lastRowIndex', (element) => parseInt(element.innerText, 10))).to.equal(3);
-        expect(await page.$eval('#totalRowsCount', (element) => parseInt(element.innerText, 10))).to.equal(3);
+        await expectInnerText(page, '#firstRowIndex', '1');
+        await expectInnerText(page, '#lastRowIndex', '3');
+        await expectInnerText(page, '#totalRowsCount', '3');
     });
 
     it('can set how many lhcPeriods is available per page', async () => {
         await goToPage(page, 'lhc-period-overview');
-        await waitForTimeout(500);
+
         // Expect the amount selector to currently be set to 10 (because of the defined page height)
-        const amountSelectorButton = await page.$('.dropup button');
-        const amountSelectorButtonText = await amountSelectorButton.evaluate((element) => element.innerText);
-        await waitForTimeout(300);
-        expect(amountSelectorButtonText.trim().endsWith('11')).to.be.true;
+        await expectInnerText(page, '.dropup button', 'Rows per page: 11 ');
 
         // Expect the dropdown options to be visible when it is selected
+        const amountSelectorButton = await page.$('.dropup button');
         await amountSelectorButton.evaluate((button) => button.click());
-        await waitForTimeout(100);
+        await page.waitForSelector('.dropup');
         const amountSelectorDropdown = await page.$('.dropup');
         expect(Boolean(amountSelectorDropdown)).to.be.true;
 
         // Expect the amount of visible lhcfills to reduce when the first option (5) is selected
         const menuItem = await page.$('.dropup .menu-item');
         await menuItem.evaluate((button) => button.click());
-        await waitForTimeout(100);
+        await page.waitForSelector('table tbody tr:nth-child(3)');
 
         const tableRows = await page.$$('table tr');
         expect(tableRows.length - 1).to.equal(3);
```

Comment:
You can probably wait for table length

Reply:
I can't because the length of the table is always 3 no matter how you filter, this causes random failure sometimes

Reply:
I don't follow. You wait for `nth-child(3)` then you check the table length to be 3, why would waitForTableLength fail?

Reply:
Because this test doesn't test what it should. First the page is opened and the default items per page is 11 (because of the page height), while the actual length of the table is 3 items (only 3 in seeders). Then the pagination is set to 5 while there are only 3 rows in the table anyways so that doesn't make sense. I changed the test to now expect that the value of the pagination is 5 and the table length is 3 and it doesn't fail anymore. I did this for all similar tests